### PR TITLE
feat: implement BASEFEE opcode

### LINKED ISSUE #54: Opcode: BASEFEE

### PR DESCRIPTION
**Pull request by ricomateo**
Closes #54 
This PR implements code generation for [BASEFEE](https://www.evm.codes/#48?fork=cancun) opcode.

### CODE REVIEW

File: tests/evm.rs [hunk 1]
```diff
@@ -455,6 +455,33 @@ fn callvalue_stack_overflow() {
     run_program_assert_halt(program, env);
 }
 
+#[test]
+fn basefee() {
+    let program = vec![Operation::Basefee];
+    let mut env = Env::default();
+    let basefee = 10_u8;
+    let expected_result = BigUint::from(basefee);
+    env.block.basefee = ethereum_types::U256::from(basefee);
```
Comment (JulianVentura) [inline tests/evm.rs:464]: Please, import `u256` type from `src/primitives.rs`

**Comment by JulianVentura** [comment]
LGTM

File: README.md [hunk 2]
```diff
@@ -45,6 +45,7 @@ An EVM-bytecode to machine-bytecode compiler using MLIR and LLVM.
 1. (0x36) CALLDATASIZE
 1. (0x37) CALLDATACOPY
 1. (0x38) CODESIZE
+1. (0x48) BASEFEE
```
Comment (MegaRedHand) [inline README.md:48]: Can you order these?

File: src/syscall.rs [hunk 3]
```diff
@@ -311,6 +311,12 @@ impl<'c> SyscallContext<'c> {
         let log = Log { data, topics };
         self.inner_context.logs.push(log);
     }
+
+    #[allow(improper_ctypes)]
```
Comment (MegaRedHand) [inline src/syscall.rs:315]: Please remove this

File: src/syscall.rs [hunk 4]
```diff
@@ -311,6 +311,12 @@ impl<'c> SyscallContext<'c> {
...
+    pub extern "C" fn get_basefee(&self, basefee: &mut U256) {
```
Comment (MegaRedHand) [inline src/syscall.rs:316]: Can we rename it to `store_in...` like the other syscalls?

**Review by MegaRedHand — APPROVED** [review]
LGTM

**Pull request merged**